lib/rules/arrow-body-style.js

```diff
@@ -245,7 +263,24 @@ module.exports = {
         }
 
         return {
-            "ArrowFunctionExpression:exit": validate
+            "BinaryExpression[operator='in']"() {
+                let info = funcInfo;
```

Comment:
@mdjermanovic just a doubt. 

why referring `funcInfo` to `info` is working but instead using `funcInfo` alone in the while loop is not working.

Reply:
I'm not sure I understand, what would the code look like?

Reply:
my doubt is like why without the `info` variable , using `funcInfo` instead in the while loop is not working. 
I tried changing that and most of the test cases failed.

Reply:
`funcInfo` is data about the arrow function that's being traversed at the given moment. If we change it before `"ArrowFunctionExpression:exit"`, `function validate` will have wrong data (or no data if it's `null`).

Reply:
I meant , changing this https://github.com/eslint/eslint/pull/13228/files#diff-c2240f0cdfc25e416753260c2a767fdeR267-R273

with

```js
"BinaryExpression[operator='in']"() {
         while (funcInfo) {
                    funcInfo.hasInOperator = true;
                    funcInfo= funcInfo.upper;
                }
            },
```

is producing wrong result, just curious to know why !

Reply:
It will become `null` at the end of the loop, so `if (astUtils.isOpeningBraceToken(firstValueToken) || blockBody[0].argument.type === "SequenceExpression" || (funcInfo.hasInOperator && isInsideForLoopInitializer(node)))` in `function validate` will throw error.

Reply:
Okk, 
so we are using the temp variable to change the property of the `funcInfo` as it is being referred there and even if `info` becomes `null`, `funcInfo` will have the non `null` value of the previous iteration of the while loop .

something like this right ?

Reply:
yes, it's just a temp variable that helps to set `hasInOperator = true` for all currently "active" arrow functions, without changing anything else.

Reply:
cool, thanks a lot 💯